[native] Plan conversion fails: Constant vector cannot wrap Constant vector

```
select 
    case when date(from_unixtime(orderkey) AT TIME ZONE 'UTC') = date('1970-01-01')
    then null
    else from_unixtime(orderkey) AT TIME ZONE 'UTC'
    end
from orders

Caused by: VeloxRuntimeError: base->BaseVector::encoding() != VectorEncoding::Simple::CONSTANT (CONSTANT vs. CONSTANT) Constant vector cannot wrap Constant vector
```

CC: @amitkdutta @spershin @zacw7

--- a/presto-native-execution/presto_cpp/main/types/PrestoToVeloxExpr.cpp
+++ b/presto-native-execution/presto_cpp/main/types/PrestoToVeloxExpr.cpp
@@ -429,8 +429,7 @@ std::shared_ptr<const ConstantTypedExpr> VeloxExprConverter::toVeloxExpr(
       auto valueVector =
           protocol::readBlock(type, pexpr->valueBlock.data, pool_);
       return std::make_shared<ConstantTypedExpr>(
-          std::make_shared<velox::ConstantVector<velox::ComplexType>>(
-              pool_, 1, 0, valueVector));
+          velox::BaseVector::wrapInConstant(1, 0, valueVector));
     }
     default: {
       const auto value = getConstantValue(type, pexpr->valueBlock);

